src/shims/foreign_items.rs

```diff
@@ -102,6 +113,49 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    /// Lookup the body of a function that has `link_name` as the symbol name.
+    fn lookup_exported_symbol(
+        &self,
+        abi: Abi,
+        link_name: Symbol,
+    ) -> InterpResult<'tcx, Option<&'mir mir::Body<'tcx>>> {
+        let this = self.eval_context_ref();
+        let tcx = this.tcx.tcx;
+        let mut instance_and_crate: Option<(ty::Instance<'_>, CrateNum)> = None;
+        for cnum in iter::once(LOCAL_CRATE).chain(tcx.crates().iter().copied()) {
+            // FIXME: Do we need to check `SymbolExportLevel` (the `_` below)?
+            for &(symbol, _) in tcx.exported_symbols(cnum) {
```

Comment:
So, on each call we do a linear search through al the symbols? That seems rather inefficient. Is there a `DefId` or something like that we can cache?
(Also okay to leave this as future work, just add a FIXME then please.)

Reply:
I *think* I implemented a cache for this (as `FxHashMap<Symbol, Instance<'tcx>>` in `Evaluator`), but I don't know how to test the cache works.

Reply:
Yeah that's the thing with caches, isn't it? They don't *do* anything other than make stuff go faster.^^